clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -971,16 +908,31 @@ static Expected<StringRef> linkDevice(ArrayRef<StringRef> InputFiles,
   InputFilesVec.clear();
   InputFilesVec.emplace_back(*LinkedFile);
 
-  // Get SYCL device library files
   // Gathering device library files
   SmallVector<std::string, 16> DeviceLibFiles;
   if (Error Err = sycl::getSYCLDeviceLibs(DeviceLibFiles, Args))
     reportError(std::move(Err));
-  SmallVector<std::string, 16> UnbundledDeviceLibFiles;
-  if (Error Err = sycl::unbundleSYCLDeviceLibs(DeviceLibFiles,
-                                               UnbundledDeviceLibFiles, Args))
-    reportError(std::move(Err));
-  for (auto &File : UnbundledDeviceLibFiles)
+  const llvm::Triple Triple(Args.getLastArgValue(OPT_triple_EQ));
+  SmallVector<std::string, 16> ExtractedDeviceLibFiles;
+  for (auto &File : DeviceLibFiles) {
```

Comment:
Could you please move this loop in a dedicated function?

Reply:
Is there a reason why we would want to do that? I do not see any benefits as we do not need this functionality elsewhere....Thanks

Reply:
My rationale was that the function name `linkDevice` suggests that some `linking` is happening here while this particular loop is performing some `extraction` rather than `linking`. In order to understand that a reader will have to make a cognitive effort to read the actual code.

Anyway, I don't insist that it should be a blocker.